server/src/main/java/io/crate/execution/engine/aggregation/impl/average/AverageAggregation.java

```diff
@@ -58,148 +55,58 @@
 import io.crate.types.LongType;
 import io.crate.types.ShortType;
 
-public class AverageAggregation extends AggregationFunction<AverageAggregation.AverageState, Double> {
+public class AverageAggregation extends AggregationFunction<AverageState, Double> {
 
     public static final String[] NAMES = new String[]{"avg", "mean"};
     public static final String NAME = NAMES[0];
 
     static {
-        DataTypes.register(AverageStateType.ID, in -> AverageStateType.INSTANCE);
+        DataTypes.register(IntegralAverageStateType.ID, in -> IntegralAverageStateType.INSTANCE);
+        DataTypes.register(FractionalAverageStateType.ID, in -> FractionalAverageStateType.INSTANCE);
     }
 
-    static final List<DataType<?>> SUPPORTED_TYPES = Lists2.concat(
-        DataTypes.NUMERIC_PRIMITIVE_TYPES, DataTypes.TIMESTAMPZ);
+    static final List<DataType<?>> SUPPORTED_INTEGRAL_TYPES = List.of(DataTypes.TIMESTAMPZ, DataTypes.BYTE, DataTypes.SHORT, DataTypes.INTEGER, DataTypes.LONG);
+    static final List<DataType<?>> SUPPORTED_FRACTIONAL_TYPES = List.of(DataTypes.FLOAT, DataTypes.DOUBLE);
```

Comment:
```suggestion
    static final List<DataType<?>> INTEGRAL_TYPES = List.of(DataTypes.TIMESTAMPZ, DataTypes.BYTE, DataTypes.SHORT, DataTypes.INTEGER, DataTypes.LONG);
    static final List<DataType<?>> FRACTIONAL_TYPES = List.of(DataTypes.FLOAT, DataTypes.DOUBLE);
```

Not sure if the `SUPPORTED_` prefix adds much here. There isn't a `UNSUPPORTED_INTEGRAL_TYPES` list, so the distinction/clarification seems unnecessary.